Offices/Offices.Data/Providers/MongoDbContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using MongoDB.Driver;
+using Offices.Data.Entities;
+
+namespace Offices.Data.Providers
+{
+    public class MongoDbContext<T> where T : BaseEntity
+    {
+        public readonly IMongoCollection<T> Entities;
```

Comment:
You should not create a separate context for each collection in mongodb. Context is used for a unified access to the database, so put all the collections to the same context without parametrizing it

Reply:
Fixed